api/src/test/java/io/strimzi/api/kafka/model/KafkaMirrorMaker2CrdIT.java

```diff
@@ -51,7 +51,7 @@ void testKafkaMirrorMaker2WithMissingRequired() {
             KubeClusterException.class,
             () -> createDeleteCustomResource("KafkaMirrorMaker2-with-missing-required-property.yaml"));
 
-        assertMissingRequiredPropertiesMessage(exception.getMessage(), "connectCluster", "clusters.alias", "sourceCluster", "targetCluster");
+        assertMissingRequiredPropertiesMessage(exception.getMessage(), "spec.connectCluster", "spec.clusters[2].alias", "sourceCluster", "targetCluster");
```

Comment:
I'm using v1.25.0 to test.

Reply:
Should I change the Kube version to avoid those errors? 🤔

Reply:
I'm running these against 1.25 as well. And I do not have any issues like that. They are all passing on the CIs as well. So that seems weird - like something in your environment is braking them. What is your `kubectl` version as some of these use `kubectl`

Reply:
```
kubectl version                                                                                                                                                                                                                   

WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.0", GitCommit:"a866cbe2e5bbaa01cfd5e969aa3e033f3282a8a2", GitTreeState:"clean", BuildDate:"2022-08-23T17:44:59Z", GoVersion:"go1.19", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.0", GitCommit:"a866cbe2e5bbaa01cfd5e969aa3e033f3282a8a2", GitTreeState:"clean", BuildDate:"2022-08-23T17:38:15Z", GoVersion:"go1.19", Compiler:"gc", Platform:"linux/amd64"}
```

Reply:
Turns out I had an old `kubectl`. With Kubectl 1.25 it fails for me as well. Sorry. That said, I do not think the fixes aou did will work on the CI which uses kubectl / Kubernetes 1.19. It would be also good to separate it from this PR because it will be much easier to review but also to make sure the tests work locally but also on the CI. 

In the cases where things changed, you will need probably to use the `anyOf` instead of just replacing the existing values. If you want, I can look at it and open the PR. Or I can leave it to you. As you prefer.

Reply:
Ok that's clear, yes I will do the change if possible.
I'm just not sure to understand the following 

> In the cases where things changed, you will need probably to use the anyOf instead of just replacing the existing values

I can see that `anyOf` [is already used here](https://github.com/strimzi/strimzi-kafka-operator/blob/5bb792252b92338e9f4f0951d55d322be0b62769/api/src/test/java/io/strimzi/api/kafka/model/AbstractCrdIT.java#L116)
You mean add other `containsStringIgnoringCase` statements to match the new patterns?

Reply:
Yes, makes sense, I'm on it. I will push a fix asap. 
Do I need to do it in a separate branch/PR or on this one?

Reply:
I think that separate branch / PR will be better, because it would make it easier to test and review. For example, you will see if it passes in the CI without being affected by the ACL changes etc.

Reply:
Indeed!

Reply:
Here it is :)

Reply:
I forgot to put the link sorry https://github.com/strimzi/strimzi-kafka-operator/pull/7433